docs/EN/operate/build.md

```diff
@@ -172,90 +198,136 @@ Sometimes, specific branches are offered for testing. Any desired branch can be
-- Select “Apply Stash After Operation”
-- Click “Stash and Pull”
+```
+git stash pop
+```
 
-The Trio code is now updated. That was easy. If you used the build script that automatically creates your ConfigOverride file, the targets are already signed, and you are ready to build the Trio app on your phone.
+If you made customizations in any of the submodules, the `recurse` command might fail. Follow the directions in [Fix Submodule Errors](#fix-submodule-errors).
 
-The following section is only for those who prefer to use the command line interface for `git`. Skip ahead to [Verify Trio Version](#verify-trio-version).
+If there were no errors, you are ready to build.
 
-### Update Trio with CLI
+### Fix Submodule Errors
 
-Use finder to locate the directory where the BuildTrio script saved the code. The directory is named after the branch with the date and time for the download:
+If a submodule gave an error in response to the `git pull --recurse` command:
 
-* Released (main) branch: Downloads/BuildTrio/Trio_main-[date-time]/Trio
-    * Example: ~/Downloads/BuildTrio/Trio_main-220122-1352/Trio
-* Development (dev) branch: Downloads/BuildTrio/Trio_dev-[date-time]/Trio
-    * Example: ~/Downloads/BuildTrio/Trio_dev-220108-1827/Trio
+* Read which submodule failed to `checkout`
+* Change directory to that module: `cd modulename`
+* Stash your changes: `git stash`
+* Return to the Trio folder and try again: `cd ..; git submodule update`
 
-Use finder to open a Terminal window at the Trio directory by right-clicking on it and selecting `New Terminal at Folder`. 
+If another submodule fails to checkout, repeat the steps for that `modulename`.
 
-Copy each line below and paste it into the terminal window one at a line and hit enter for each line:
+If the modification is one you want to keep, you must restore it:
+ 
+* Change directory to that module: `cd modulename`
+* Restore your changes: `git stash pop`
+* Return to the Trio folder: `cd ..`
+
+The Trio code is now updated.
+
+If Xcode is not open, you can open it by typing `xed .` in the Trio folder of your terminal window.
+
+You can now build the updated Trio app on your phone.
+
+## Xcode Behaviors
+
+You can add Behaviors to your version of Xcode. These custom Behaviors only need to be added once.
+
+You will create a shell script and add it to Xcode.
+
+* `Open Terminal` will open a terminal in the workspace or project folder currently in use by Xcode
+
+### Prepare the Shell Script
+
+Open a new terminal:
+
+Step 1: Create a folder to store your shell scripts by copying and pasting the next line (only do this one time or you will get an error message):
 
 ```
-git stash
+mkdir ~/scripts
 ```
 
+Step 2: Copy and paste this command into your terminal to prepare the `Open Terminal` shell script:
+
 ```
-git fetch
-git pull --recurse
+open -a TextEdit ~/scripts/open_terminal.sh
```

Comment:
This command fails because `The file /Users/username/scripts/open_terminal.sh does not exist.`

Also, looks like `-e` does the same thing as `-a TextEdit`, or `-t` would open in the default text editor if that isn't TextEdit

So maybe update the line to?:
`touch ~/scripts/open_terminal.sh; open -e ~/scripts/open_terminal.sh`

Reply:
Thanks - added your fix. I forgot to delete everything and start fresh after I figured out an easier way to do it than what I originally found posted on stackoverflow. Did that now and it works with your modification.